test/compress/issue-1443.js

```diff
@@ -0,0 +1,53 @@
+unsafe_undefined: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {}
+    input: {
+        function f(undefined) {
+            var a;
+            return function() {
+                console.log(typeof a == "undefined");
+            }
+        }
+    }
+    expect: {
+        function f(n) {
+            var o;
+            return function() {
+                console.log(n===o);
+            }
+        }
+    }
+}
+
+keep_fnames: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {
+        keep_fnames: true
+    }
+    input: {
+        function f(undefined) {
+            return function() {
+                function n(a) {
+                    return a * a;
+                }
+                return typeof n == "undefined";
+            };
+        }
+    }
+    expect: {
+        function f(r) {
+            return function() {
+                function n(n) {
+                    return n * n;
+                }
+                return r === n;
```

Comment:
This is [utilising the rule](https://github.com/mishoo/UglifyJS2/blob/0610c020b1544820be9898a285ab6c9066490552/lib/compress.js#L2495-L2518) `typeof a == "undefined"` :arrow_right: `a === undefined`, specifically [this line](https://github.com/mishoo/UglifyJS2/blob/0610c020b1544820be9898a285ab6c9066490552/lib/compress.js#L2514) turns `==` into `===`.

`evaluate` is not enabled in this test case, so `typeof function(){}` wouldn't get optimised any further.